lib/class-wp-duotone-gutenberg.php

```diff
@@ -0,0 +1,367 @@
+<?php
+/**
+ * WP_Duotone_Gutenberg class
+ *
+ * @package gutenberg
+ * @since 6.3.0
+ */
+
+/**
+ * Manages which duotone filters need to be output on the page.
+ *
+ * @access public
+ */
+class WP_Duotone_Gutenberg {
+	/**
+	 * An array of Duotone presets from global, theme, and custom styles.
+	 *
+	 * Example:
+	 * [
+	 *      'blue-orange' =>
+	 *          [
+	 *              'slug'  => 'blue-orange',
+	 *              'colors' => [ '#0000ff', '#ffcc00' ],
+	 *          ]
+	 *      ],
+	 *      …
+	 * ]
+	 *
+	 * @since 6.3.0
+	 * @var array
+	 */
+	private static $global_styles_presets = array();
+
+	/**
+	 * An array of block names from global, theme, and custom styles that have duotone presets. We'll use this to quickly
+	 * check if a block being rendered needs to have duotone applied, and which duotone preset to use.
+	 *
+	 * Example:
+	 *  [
+	 *      'core/featured-image' => 'blue-orange',
+	 *       …
+	 *  ]
+	 *
+	 * @since 6.3.0
+	 * @var array
+	 */
+	private static $global_styles_block_names = array();
+
+	/**
+	 * An array of Duotone SVG and CSS output needed for the frontend duotone rendering based on what is
+	 * being output on the page. Organized by a slug of the preset/color group and the information needed
+	 * to generate the SVG and CSS at render.
+	 *
+	 * Example:
+	 *  [
+	 *      'blue-orange' => [
+	 *          'slug'  => 'blue-orange',
+	 *          'colors' => [ '#0000ff', '#ffcc00' ],
+	 *      ],
+	 *      'wp-duotone-000000-ffffff-2' => [
+	 *          'slug' => 'wp-duotone-000000-ffffff-2',
+	 *          'colors' => [ '#000000', '#ffffff' ],
+	 *      ],
+	 * ]
+	 *
+	 * @since 6.3.0
+	 * @var array
+	 */
+	private static $output = array();
+
+	const CSS_VAR_PREFIX = '--wp--preset--duotone--';
```

Comment:
One last nit. I don't think this necessarily has to be `private` as just a constant string, but PHP defaults to `public` and it wasn't explicitly labeled as such.

```suggestion
	/**
	 * Prefix used for generating and referencing duotone CSS custom properties.
	 */
	private const CSS_VAR_PREFIX = '--wp--preset--duotone--';
```